sig-architecture/api-review-process.md

```diff
@@ -0,0 +1,245 @@
+
+Because expert reviewer bandwidth is extremely limited, the process provides a curated backlog with highest priority issues at the top. While this does mean some changes may be delayed in favor of other higher priority ones, this will help maintain critical project velocity, transparency, and equilibrium. Ideally, those whose API review priority is shifted in a release-impacting way will be proactively notified by the reviewers.
+
+# Goals of the process
+
+* Provide an easily-navigable process so all parties understand their roles, responsibilities, and expectations
+
+* Protect Kubernetes APIs from disruptive, inconsistent, or destabilizing changes
+
+* Respect, gate, and expand expert reviewer bandwidth while maintaining consistent flow across the process
+
+* Provide transparency, including clear feedback and path forward for API contributors
+
+* Maintain the high standards of the project, including positive user interactions with APIs
+
+* Provide review regardless of method of API defininition (built-in, Extension API Server, or Custom Resource Definition)
+
+* Provide review over both tightly coupled external projects and in-tree API changes.  
+
+* Provide a platform for commenting on popular extensions' APIs, so as to encourage the spread of good API practices throughout the ecosystem
+
+# Non-goals of the process
+
+* Creation or maintenance of the API standards documentation
+
+* Being a "rubber stamp" for changes
+
+# Process Description
+
+![image alt text](image_0.png)
+
+
+## Intake Pre-review Checklist
+
+* The change/addition has been reviewed by the appropriate sub-project stakeholders and SIG chairs as needed, or else the request may be rejected pending initial review
+
+* A KEP has been created if introducing:
+
+    * Any new resource type
+
+    * Any new version of a stable API
+
+    * Any major new functionality added to a stable API as defined by SIG Architecture and the API Reviewers
+
+## What APIs need to be reviewed?
+
+* What are the kind of reviews?
+
+    * **mandatory**: The changer is expected to submit the changes to API review, and the change is blocked on the API reviewer approval 
+
+    * **voluntary**: Changer can request a review, or a 3rd party can nominate the API for review.  The review comments are recommendations.
+
+* Which projects are in scope?
+
+    * github.com/kubernetes/kubernetes, and any kubernetes-* org projects it depends on are required to be reviewed.   (mandatory)
+
+    * any other github projects in the kubernetes-* orgs (informational, for now - may make mandatory in future)
+
+    * Any projects that produce APIs (including CRDs) that are intended to work with kubectl and/or kube-apiserver. (informational - intent is to ensure consistent user experience across the Kubernetes ecosystem)
```

Comment:
Any kubernetes-style API that uses a `*.k8s.io` or `*.kubernetes.io` name, e.g. "storage.k8s.io" (mandatory)